### PR TITLE
Dashboard refactor

### PR DESCRIPTION
**Pull request by tingaloo**
Refactor Dashboard
Upgrade react-bootstrap
Fix one of few bugs in Compare tab (works fine on prod site, fails on develop)

## Description
Broke down dashboard into multiple containers/components. Trying to separate concerns to make things easier to reason. Currently works as intended.

Made a fix on compare tab where somebody selects a race but nothing returns (only in dev). There is still an issue on when user picks a date.

I'll keep dashboardComponent around until this is approved. It currently works but the data passing, etc can definitely be more streamlined.

Upgraded react-bootstrap to 0.30 to remove deprecation errors, but makes a few breaking changes with forms and input components. 

could use eyes from @romoy 

## Motivation and Context
Dashboard was >300 lines

## How Has This Been Tested?
In development.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I've talked through the changes with a team lead
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by romoy** [comment]
Ah yea!! @tingaloo I will check it out.

**Review by mkalish — APPROVED** [review]

File: src/components/search/raceSearchComponent.jsx [hunk 1]
```diff
@@ -110,3 +113,11 @@ class RaceSearch extends React.Component {
 }
 
 export default RaceSearch;
+
+// <Input ref="raceType" type="select" label="Select a race type" onChange={this._handleRaceSelection.bind(this)}>
```
Comment (romoy) [inline src/components/search/raceSearchComponent.jsx]: could this be removed?

File: src/components/dashboard/dashboardComponent.jsx [hunk 2]
```diff
@@ -122,7 +121,7 @@ class Dashboard extends React.Component {
     }
 
     componentWillMount() {
-        let that = this;
+        // let that = this;
```
Comment (romoy) [inline src/components/dashboard/dashboardComponent.jsx]: could this be removed?

**Comment by tingaloo** [comment]
@romoy woops, left some comments. It should be clean now.

**Comment by romoy** [comment]
@tingaloo It's merged!!! 👍  Thanks for the break up!!

**Comment by romoy** [comment]
You should have the power to do future merge after an approved pull.

**Pull request merged**